### PR TITLE
Replace SNAPCRAFT_PART_INSTALL in the part attributes.

### PR DESCRIPTION
**Pull request by josepht**
LP:#1628915

### CODE REVIEW

**Comment by josepht** [comment]
Any suggestions for the integration test?  My example that I tested with was just a make plugin with `make-parameters` set to ```$SNAPCRAFT_SNAP_INSTALL```

**Comment by sergiusens** [comment]
@josepht a cmake one would be more complete

**Comment by kyrofa** [comment]
Yeah looks good to me too, but I agree that tests are needed.

File: integration_tests/test_lifecycle.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+import integration_tests
+
+
+class LifecycleTestCase(integration_tests.TestCase):
```
Comment (sergiusens) [inline integration_tests/test_lifecycle.py]: ca you add this to `integration_tests.test_environment` instead

File: integration_tests/snaps/simple-cmake-replace/test.c [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#include <stdio.h>
+int main() {
+    printf("It's a CMake world\n");
```
Comment (sergiusens) [inline integration_tests/snaps/simple-cmake-replace/test.c]:
```c
    printf("When I build I was installed to PART_INSTALL\n");
```

File: integration_tests/snaps/simple-cmake-replace/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+description: a longer description
+icon: icon.png
+confinement: strict
+
+build-packages: [gcc, libc6-dev]
+
+parts:
+  cmake-project:
+    plugin: cmake
+    configflags: ['-DCMAKE_INSTALL_PREFIX=$SNAPCRAFT_PART_INSTALL']
```
Comment (sergiusens) [inline integration_tests/snaps/simple-cmake-replace/snapcraft.yaml]:
```yaml
    configflags: ['-DPART_INSTALL=$SNAPCRAFT_PART_INSTALL']
```

File: integration_tests/test_lifecycle.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        project_dir = 'simple-cmake-replace'
+        self.run_snapcraft('stage', project_dir)
+
+        binary_output = self.get_output_ignoring_non_zero_exit(
+            os.path.join(
+                'stage', os.path.relpath(os.getcwd(), '/'),  project_dir,
+                'parts', 'cmake-project', 'install', 'bin',
+                'simple-cmake-replace'),
+            cwd=project_dir)
+        self.assertEqual("It's a CMake world\n", binary_output)
```
Comment (sergiusens) [inline integration_tests/test_lifecycle.py]: Don't use the `non_zero_exit` version.
```python
   os.path.join('stage', 'bin', 'simple-cmake-replace')
self.assertEqual("When I built I was installed to {}\n", binary_output)
```

File: integration_tests/snaps/simple-cmake-replace/test.c (hunk 2 above)
Comment (josepht) [inline integration_tests/snaps/simple-cmake-replace/test.c]: Done with some minor editing :)

File: integration_tests/snaps/simple-cmake-replace/snapcraft.yaml (hunk 3 above)
Comment (josepht) [inline integration_tests/snaps/simple-cmake-replace/snapcraft.yaml]: Done.

File: integration_tests/test_lifecycle.py (hunk 1 above)
Comment (josepht) [inline integration_tests/test_lifecycle.py]: Done.

File: integration_tests/test_lifecycle.py (hunk 4 above)
Comment (josepht) [inline integration_tests/test_lifecycle.py]: Done.

File: integration_tests/snaps/simple-cmake-replace/snapcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+description: a longer description
+icon: icon.png
+confinement: strict
+
+build-packages: [gcc, libc6-dev]
+
+parts:
+  cmake-project:
+    plugin: cmake
+    configflags: ['-DCMAKE_INSTALL=$SNAPCRAFT_PART_INSTALL']
```
Comment (sergiusens) [inline integration_tests/snaps/simple-cmake-replace/snapcraft.yaml]: this should be `PART_INSTALL` :wink:

File: integration_tests/snaps/simple-cmake-replace/CMakeLists.txt [hunk 6]
```diff
@@ -1,4 +1,9 @@
 cmake_minimum_required(VERSION 3.2)
 project(simple-cmake-replace C)
+configure_file(
+    "${PROJECT_SOURCE_DIR}/config.h.in"
+    "${PROJECT_SOURCE_DIR}/config.h"
```
Comment (sergiusens) [inline integration_tests/snaps/simple-cmake-replace/CMakeLists.txt]: This should be `PROJECT_BINARY_DIR` though, right?

File: integration_tests/snaps/simple-cmake-replace/CMakeLists.txt [hunk 7]
```diff
@@ -1,4 +1,9 @@
...
+    )
+include_directories("${PROJECT_SOURCE_DIR}")
```
Comment (sergiusens) [inline integration_tests/snaps/simple-cmake-replace/CMakeLists.txt]: include `BINARY` instead.

**Comment by sergiusens** [comment]
I might be wrong @josepht, wait for @kyrofa to confirm.

File: integration_tests/snaps/simple-cmake-replace/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.2)
+project(simple-cmake-replace C)
+configure_file(
+    "${PROJECT_SOURCE_DIR}/config.h.in"
+    "${PROJECT_SOURCE_DIR}/config.h"
+    )
```
Comment (kyrofa) [inline integration_tests/snaps/simple-cmake-replace/CMakeLists.txt]: I think you can get around this by simply doing `add_definitions(-DPART_INSTALL=${PART_INSTALL})`. More info [here](https://cmake.org/cmake/help/v3.0/command/add_definitions.html).

Comment (josepht) [inline integration_tests/snaps/simple-cmake-replace/CMakeLists.txt]: That works.  Thanks for the suggestion.

Comment (sergiusens) [inline integration_tests/snaps/simple-cmake-replace/CMakeLists.txt]: this was what I wanted! :-)

**Pull request merged**